File: app/src/main/java/com/vincent/githubusersearch/bases/BasePageConfig.java
package com.vincent.githubusersearch.bases;

import androidx.paging.PagedList;

public abstract class BasePageConfig {

    protected abstract int getPageSize();

    protected abstract int getInitialPageKey();

    protected PagedList.Config getConfig() {
        return new PagedList.Config.Builder()
                .setPageSize(getPageSize())
                .setInitialLoadSizeHint(getPageSize())
                .setEnablePlaceholders(false)
                .build();
    }
}

File: app/src/main/java/com/vincent/githubusersearch/bases/BasePageConfig.kt
package com.vincent.githubusersearch.bases

import androidx.paging.PagedList

/**
 * For Paging config.
 */
abstract class BasePageConfig {

    protected abstract fun getPageSize(): Int

    protected abstract fun getInitialPageKey(): Int

    protected val config: PagedList.Config
        get() = PagedList.Config.Builder()
                .setPageSize(getPageSize())
                .setInitialLoadSizeHint(getPageSize())
                .setEnablePlaceholders(false)
                .build()
}